pwiz_tools/Skyline/Model/ImportPeakBoundaries.cs

```diff
@@ -100,26 +101,57 @@ public enum Field { modified_peptide, filename, apex_time, start_time, end_time,
 
         // ReSharper disable LocalizableElement
 
+        private static string[] _peptide_synonyms; // Cache for performance reasons (L10N stuff is expensive)
         public static string[] PEPTIDE_SYNONYMS
         {
             get
             {
-                return new[] 
+                if (_peptide_synonyms == null)
                 {
-                    "PeptideModifiedSequence", "ModifiedSequence", "FullPeptideName", "EG.ModifiedSequence",
-                    ColumnCaptions.PeptideModifiedSequence, ColumnCaptions.ModifiedSequence
-                };
+                    var headers = new List<string>()
+                    {
+                        "PeptideModifiedSequence", "ModifiedSequence", "FullPeptideName", "EG.ModifiedSequence",
+                    };
+                    // Also recognize column captions in all supported display languages
+                    var currentUICulture = Thread.CurrentThread.CurrentUICulture;
+                    foreach (var culture in CultureUtil.AvailableDisplayLanguages())
+                    {
+                        Thread.CurrentThread.CurrentUICulture = culture;

```

Comment:
There is disposable CurrentCultureSetter that does this.

Reply:
Thanks. It seemed like a lot of code duplication.